test/post.test.js

```diff
@@ -92,7 +92,22 @@ describe('Post data class tests', () => {
     expect(() => createPostWithFeed({})).toThrow();
     expect(() => createPostWithFeed(feed)).not.toThrow();
   });
-
+  test('Post constructor should throw if invalid dates is passed', () => {
+    const createPostWithDates = (datePublished, dateUpdated) =>
+      new Post(data.title, data.html, datePublished, dateUpdated, data.url, data.guid, feed);
+    expect(() =>
+      createPostWithDates(
+        new Date('Thu, 20 Nov 2014 18:59:18 UTC'),
+        new Date('Fri, 28 Nov 2014 18:59:18 UTC')
+      )
+    ).not.toThrow();
+    expect(() => createPostWithDates(new Date('Thu, 20 Nov 2014 18:59:18 UTC'), 23)).not.toThrow();
+    expect(() => createPostWithDates(23, new Date('Thu, 20 Nov 2014 18:59:18 UTC'))).toThrow();
```

Comment:
I was thinking this line wouldn't throw since we have a fallbackDate

Reply:
@c3ho thats right,... I tried to use .not.toThrow() but it fails.
```
function ensureDate(date, fallbackDate) {
  if (
    date &&
    (Object.prototype.toString.call(date) === '[object String]' ||
      (Object.prototype.toString.call(date) === '[object Date]' && !Number.isNaN(date)))
  ) {
    return new Date(date);
  }
  if (Object.prototype.toString.call(fallbackDate) === '[object Date]') {
    return new Date(fallbackDate);
  }

  throw new Error(`post has an invalid date: ${date}'`);
}
```
![image](https://user-images.githubusercontent.com/36198917/99155509-711aff80-2686-11eb-9a68-821ff0d27226.png)

Reply:
I just noticed if I dod this:       createPostWithDates('zzz', new Date('Thu, 20 Nov 2014 18:59:18 UTC'))
then it throws but when I put a number it fails.


![image](https://user-images.githubusercontent.com/36198917/99155599-5a28dd00-2687-11eb-8f32-a7cca618ed55.png)
![image](https://user-images.githubusercontent.com/36198917/99155603-63b24500-2687-11eb-8ab0-ef3a24d41ff8.png)

Reply:
'zzz' is a string clears the first if statement so it returns `new Date('zzz')`

Reply:
Right but why would it throw with     
expect(() => createPostWithDates('zzz', new Date('Thu, 20 Nov 2014 18:59:18 UTC'))).toThrow();
but not
expect(() => createPostWithDates(23, new Date('Thu, 20 Nov 2014 18:59:18 UTC'))).toThrow();
For the purpose of this issue should I just change it to that and upload the final code? If so do you want me to make a new branch or push to the [issue-1323](https://github.com/joey2031/telescope/tree/issue-1323) branch? Thanks.

Reply:
It should be this branch, change it to following:
`expect(() => createPostWithDates('zzz', new Date('Thu, 20 Nov 2014 18:59:18 UTC'))).not.toThrow();`

Reply:
@c3ho I'm going to make a new branch form master copy and paste my code and make a PR. Thank you so much for all of your help!